html/pfappserver/root/static.alt/src/globals/pfValidators.js

```diff
@@ -30,13 +73,45 @@ export const inArray = (array) => {
   })
 }
 
-export const isDate = (value) => (!value) || value === '0000-00-00' || moment(value, 'YYYY-MM-DD', true).isValid()
+export const isDateFormat = (dateFormat) => {
+  return (0, _common.withParams)({
+    type: 'isDateFormat',
+    dateFormat: dateFormat
+  }, function (value) {
+    return !(0, _common.req)(value) || format(parse(value), dateFormat) === value || dateFormat.replace(/[a-z]/gi, '0') === value
+  })
+}
+
+export const categoryIdNumberExists = (value, component) => {
+  if (!value || !/\d+/.test(value)) return true
+  return store.dispatch('config/getRoles').then((response) => {
+    return (response.filter(role => role.category_id === value).length > 0)
+  }).catch(() => {
+    return true
+  })
+}
+
+export const categoryIdStringExists = (value, component) => {
+  if (!value || /\d+/.test(value)) return true
```

Comment:
Test should be `/^\d+$/`.

Reply:
I think we should also ignore strings starting with integers, thus we should use `/^\d+/`